src/zkevm_specs/evm_circuit/execution/create.py

```diff
@@ -0,0 +1,186 @@
+    # so the last part (GAS_COST_CODE_DEPOSIT * len(byte_code)) won't be calculated here
+    gas_left = instruction.curr.gas_left
+    gas_cost = GAS_COST_CREATE + memory_expansion_gas_cost
+    if is_create2 == 1:
+        memory_size, _ = instruction.constant_divmod(size.expr() + 31, FQ(32), N_BYTES_MEMORY_SIZE)
+        gas_cost += GAS_COST_COPY_SHA3 * memory_size
+    gas_available = gas_left - gas_cost
+
+    # Apply EIP 150
+    one_64th_gas, _ = instruction.constant_divmod(gas_available, FQ(64), N_BYTES_GAS)
+    all_but_one_64th_gas = gas_available - one_64th_gas
+    is_u64_gas = instruction.is_zero(
+        instruction.sum(instruction.rlc_encode(gas_left, N_BYTES_GAS).le_bytes[N_BYTES_GAS:])
+    )
+    callee_gas_left = instruction.select(
+        is_u64_gas,
+        instruction.min(all_but_one_64th_gas, gas_left, N_BYTES_GAS),
+        all_but_one_64th_gas,
+    )
+
+    # copy init_code from memory to bytecode
+    copy_rwc_inc, _ = instruction.copy_lookup(
+        instruction.curr.call_id,  # src_id
+        CopyDataTypeTag.Memory,  # src_type
+        instruction.next.code_hash.expr(),  # dst_id
+        CopyDataTypeTag.Bytecode,  # dst_type
+        offset,  # src_addr
+        offset + size,  # src_addr_boundary
+        FQ(1),  # dst_addr
+        size,  # length
+        instruction.curr.rw_counter + instruction.rw_counter_offset,
+    )
+    instruction.rw_counter_offset += int(copy_rwc_inc)
+
+    # CREATE:  3 pops and 1 push, stack delta = 2
+    # CREATE2: 4 pops and 1 push, stack delta = 3
+    stack_pointer_delta = 2 + is_create2
+    # Save caller's call state
+    for field_tag, expected_value in [
+        (CallContextFieldTag.ProgramCounter, instruction.curr.program_counter + 1),
+        (
+            CallContextFieldTag.StackPointer,
+            instruction.curr.stack_pointer + stack_pointer_delta,
+        ),
+        (CallContextFieldTag.GasLeft, gas_left - gas_cost - callee_gas_left),
+        (CallContextFieldTag.MemorySize, next_memory_size),
+        (
+            CallContextFieldTag.ReversibleWriteCounter,
+            instruction.curr.reversible_write_counter + 1,
+        ),
+    ]:
+        instruction.constrain_equal(
+            instruction.call_context_lookup(field_tag, RW.Write),
+            expected_value,
+        )
+
+    # Setup next call's context.
+    for field_tag, expected_value in [
+        (CallContextFieldTag.CallerId, instruction.curr.call_id),
+        (CallContextFieldTag.TxId, tx_id.expr()),
+        (CallContextFieldTag.Depth, depth.expr() + 1),
+        (CallContextFieldTag.CallerAddress, caller_address.expr()),
+        (CallContextFieldTag.CalleeAddress, contract_address.expr()),
+        (CallContextFieldTag.IsSuccess, is_success),
+        (CallContextFieldTag.IsStatic, FQ(False)),
+        (CallContextFieldTag.IsRoot, FQ(False)),
+        (CallContextFieldTag.IsCreate, FQ(False)),
+        (CallContextFieldTag.CodeHash, code_hash),
```

Comment:
```suggestion
        (CallContextFieldTag.CodeHash, code_hash.expr()),
```

Reply:
fixed in [d675ec9](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/355/commits/d675ec97ed971ec044a5bc5a40db3ab57551c629)